### PR TITLE
Cairo1 syscalls refactoring

### PR DESCRIPTION
**Pull request by maciejka**
Combines `Request`, `Response`, `syscall_exec` into a single trait which makes code more straight forward.

### CODE REVIEW

File: src/cairo_types/mod.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 pub(crate) mod bigint;
 pub(crate) mod builtins;
 pub(crate) mod dict_access;
+pub(crate) mod new_syscalls;
```
Comment (notlesh) [inline src/cairo_types/mod.rs:4]: "new" isn't very clear to me. suggestions:

* `cairo1_syscalls` (may not be strictly correct)
* `current_syscalls`
* a brief explanation inside `new_syscalls.rs` about new vs. deprecated

Comment (maciejka) [reply src/cairo_types/mod.rs:4]: There is a corresponding new_syscalls.cairo file.

File: src/execution/syscall_handler.rs [hunk 2]
```diff
@@ -59,47 +61,619 @@ impl OsSyscallHandlerWrapper {
...
+    fn write_response(
+        _response: Self::Response,
+        _vm: &mut VirtualMachine,
+        _ptr: &mut Relocatable,
+    ) -> WriteResponseResult {
+        Ok(())
+    }
+}
+
+// TODO: Keccak syscall.
```
Comment (notlesh) [inline src/execution/syscall_handler.rs:467]: Just to confirm: all of this commented out code was just moved, right?

File: src/execution/syscall_handler_utils.rs [hunk 3]
```diff
@@ -300,54 +222,83 @@ pub fn write_segment(vm: &mut VirtualMachine, ptr: &mut Relocatable, segment: Re
...
 }
 
-impl SyscallResponse for SingleSegmentResponse {
-    fn write(self, vm: &mut VirtualMachine, ptr: &mut Relocatable) -> WriteResponseResult {
-        write_segment(vm, ptr, self.segment)
-    }
+pub trait SyscallHandler {
+    type Request;
+    type Response;
+    fn read_request(vm: &VirtualMachine, ptr: &mut Relocatable) -> SyscallResult<Self::Request>;
```
Comment (notlesh) [inline src/execution/syscall_handler_utils.rs:228]: This is already an improvement, but I think we could also generate the `read_request()` and `write_response()` code. WDYT?

**Review by notlesh — APPROVED** [review]

File: src/execution/syscall_handler_utils.rs (hunk 3 above)
Comment (maciejka) [reply src/execution/syscall_handler_utils.rs:228]: you mean noop default implementation?

**Pull request merged**